src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -291,18 +291,29 @@ public override BoundNode VisitCompoundAssignmentOperator(BoundCompoundAssignmen
 
         public override BoundNode Visit(BoundNode node)
         {
-            if (_inExpressionLambda &&
-                // Ignoring BoundConversion nodes prevents redundant diagnostics
-                !(node is BoundConversion) &&
-                node is BoundExpression expr &&
-                expr.Type is TypeSymbol type &&
-                type.IsRestrictedType())
+            if (_inExpressionLambda && node is BoundExpression expression)
             {
-                Error(ErrorCode.ERR_ExpressionTreeCantContainRefStruct, node, type.Name);
+                ReportSuppressionIfNeeded(expression);
+
```

Comment:
I don't see a problem with suppression in expression trees.  Like parens, they would be applied at compile-time only.